src/utils/typeUtils.ts

```diff
@@ -4,18 +4,19 @@
 // allows any descendent properties to be omitted.
 export type RecursivePartial<T> = {
   [P in keyof T]?:
-  T[P] extends (infer U)[] ? RecursivePartial<U>[] :
-    // eslint-disable-next-line @typescript-eslint/ban-types
-    T[P] extends object ? RecursivePartial<T[P]> :
-      T[P];
+  T[P] extends Array<infer U> ? Array<U> :
```

Comment:
This no longer seems to be `RecursivePartial`-ing:
```ts
var a: RecursivePartial<{one: {two: string}[]}>;
a = {one: [{}]};
// Property 'two' is missing in type '{}' but required in type '{two: string; }'. (TS2741)
```

The `T[P] extends (infer U)[] ? RecursivePartial<U>[] :` clause should already be handling arrays; it's unclear why this is needed? If you'd like, rewriting that as `T[P] extends Array<infer U> ? Array<RecursivePartial<U>> :` might be clearer?

I do not seem to have build errors with this hunk reverted.

Reply:
I've removed all changes to `src/utils/typeUtils.ts` because they are no longer necessary because `imageAllowList` is no longer inside `kubernetes`.